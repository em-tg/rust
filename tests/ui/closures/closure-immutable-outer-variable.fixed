//@ check-pass
//@ run-rustfix

// Point at the captured immutable outer variable

fn foo(mut f: Box<dyn FnMut()>) {
    f();
}

fn main() {
    let mut y = true;
    foo(Box::new(move || y = !y) as Box<_>);
    //~^ WARNING cannot assign to `y`, as it is not declared as mutable
}
