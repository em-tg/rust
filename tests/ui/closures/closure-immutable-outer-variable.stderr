warning[E0594]: cannot assign to `y`, as it is not declared as mutable
  --> $DIR/closure-immutable-outer-variable.rs:12:26
   |
LL |     foo(Box::new(move || y = !y) as Box<_>);
   |                          ^^^^^^ cannot assign
   |
   = note: `#[warn(mut_non_mut)]` on by default
help: consider changing this to be mutable
   |
LL |     let mut y = true;
   |         +++

warning: 1 warning emitted

For more information about this error, try `rustc --explain E0594`.
